kits/cpp/src/lux/config.hpp

```diff
@@ -70,6 +70,13 @@ namespace lux {
     };
     NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(WeatherConfigs, COLD_SNAP, DUST_STORM, MARS_QUAKE, SOLAR_FLARE)
 
+    struct ActionQueueConfigs {
+        ActionQueueConfigs HEAVY;
+        ActionQueueConfigs LIGHT;
+        const ActionQueueConfigs &operator[](const std::string &name) const;
+    };
+    NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(ActionQueueConfigs, HEAVY, LIGHT)
```

Comment:
This struct will not compile for multiple reasons:
a) The struct type inside the struct itself is a level of recursion that is not allowed :D
b) The implementation for the operator overload is missing.

In case [this](https://github.com/Lux-AI-Challenge/Lux-Design-2022/blob/dde666dc9c012d263623e08ecef82516f929d8c2/kits/sample_env_cfg.json#L59) is an up to date version of what we can expect `UNIT_ACTION_QUEUE_POWER_COST` to look like the struct should be:
```cpp
    struct ActionQueueConfigs {
        int64_t HEAVY;
        int64_t LIGHT;
        int64_t operator[](const std::string &name) const;
    };
    NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(ActionQueueConfigs, HEAVY, LIGHT)
```
This addresses a).
To address b) you can add the following to `config.cpp`
```cpp
    int64_t ActionQueueConfigs::operator[](const std::string &name) const {
        if (name == "HEAVY") {
            return HEAVY;
        }
        return LIGHT;
    }
```